src/Disks/IO/ReadBufferFromRemoteFSGather.cpp

```diff
@@ -79,7 +79,7 @@ SeekableReadBufferPtr ReadBufferFromRemoteFSGather::createImplementationBuffer(c
     std::unique_ptr<ReadBufferFromFileBase> buf;
 
 #ifndef CLICKHOUSE_KEEPER_STANDALONE_BUILD
-    if (with_file_cache)
+    if (with_file_cache && settings.remote_fs_cache->isInitialized())
```

Comment:
This would make queries (that read from object storage) fail if cache fails to initialize, right? I wonder if it would be better to emit a warning instead, as queries can be made reading from object storage directly, or at least have a setting to not throw in that case

Reply:
> This would make queries (that read from object storage) fail if cache fails to initialize, right?

Yes, this behaviour will be consistent with what it currently is.

>  wonder if it would be better to emit a warning instead, as queries can be made reading from object storage directly, or at least have a setting to not throw in that case

Yes, let's add a setting, but by default let's leave the behaviour as it is currently is - e.g. throw.

Reply:
Added a setting for this